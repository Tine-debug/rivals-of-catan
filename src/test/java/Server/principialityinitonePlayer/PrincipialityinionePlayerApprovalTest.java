import java.io.OutputStream;
import java.io.PrintStream;
import java.util.ArrayList;
import java.util.List;
import java.util.Vector;
import java.util.stream.Stream;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;
import org.junit.jupiter.params.provider.ValueSource;


public class PrincipialityinionePlayerApprovalTest{
    Player player;

    int center = 2;
    int[][] regionDice = { { 2, 1, 6, 3, 4, 5 }, { 3, 4, 5, 2, 1, 6 } };

    @BeforeAll
    public static void setupall(){
        try {
            Cardstacks.loadBasicCards("cards.json");
            }    catch (Exception e) {
            System.err.println("Failed to load");
            }
    }

    @BeforeEach
    void setup(){
        player = new Player();
    }


    @Test
    void principalityinitoineplayertest1(){
        Server.pricipalityinitoneplayer(player, regionDice, center, 1);
        Approvals.verify(player.principality.printPrincipality(player));
    }

    @Test
    void principalityinitoineplayertest0(){
        Server.pricipalityinitoneplayer(player, regionDice, center, 0);
        Approvals.verify(player.principality.printPrincipality(player));
    }





}