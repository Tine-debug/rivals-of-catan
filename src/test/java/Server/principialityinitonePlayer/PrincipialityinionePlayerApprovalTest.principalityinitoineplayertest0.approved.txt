      Col 0               Col 1        Col 2                 Col 3        Col 4              
    +-------------------+------------+---------------------+------------+------------------+
 0  |                   |            |                     |            |                  |
    |                   |            |                     |            |                  |
    +-------------------+------------+---------------------+------------+------------------+
 1  | Forest (L):Lumber |            | Gold Field (A):Gold |            | Field (G):Grain  |
    | d2  1/3           |            | d1  0/3             |            | d6  1/3          |
    +-------------------+------------+---------------------+------------+------------------+
 2  |                   | Settlement | Road                | Settlement |                  |
    |                   | Center     | Center              | Center     |                  |
    +-------------------+------------+---------------------+------------+------------------+
 3  | Hill (B):Brick    |            | Pasture (W):Wool    |            | Mountain (O):Ore |
    | d3  1/3           |            | d4  1/3             |            | d5  1/3          |
    +-------------------+------------+---------------------+------------+------------------+
 4  |                   |            |                     |            |                  |
    |                   |            |                     |            |                  |
    +-------------------+------------+---------------------+------------+------------------+

Points: VP=2  CP=0  SP=0  FP=0  PP=0
